runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8201,6 +8201,11 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
                                      reloRuntime);
       }
 
+#if defined(OSX) && defined(AARCH64)
+   // Re-aquire execution permission of JIT code cache for this thread
+   // regardless of the previous protection status
+   pthread_jit_write_protect_np(1);
+#endif
```

Comment:
I suggest we're overdue for the introduction of a helper function for this (that would do nothing on platforms where it's not required). The need for it may extend to other platforms in the future: a helper function would make dealing with that much more straight-forward.

Reply:
I agree with your suggestion.  But please let me do that later with Issue #14656, not this time.
I would like to deliver this fix soon, and apply it also to the v0.35.0 branch.